Shopify/ruby-lsp
# Fix parsing erb with bare `yield`

### Motivation

Some things that are valid in erb would not be valid in a standalone ruby file. This particularly impacts `yield`.
Instead treat the provided code as being evaled

Closes #2442

### Implementation

Switch prism into eval mode. Prior work for the prism parser translator at https://github.com/ruby/prism/pull/2741

### Automated Tests

Yup

### Manual Tests

Create a new erb file with the content from the test and observe that there are no syntax errors
